pdfiumandroid/src/main/cpp/pdfiumandroid.cpp

```diff
@@ -58,12 +58,41 @@ struct rgb {
     uint8_t blue;
 };
 
+JavaVM* javaVm;
```

Comment:
So, regarding the JNI: most essential info can be found here:
https://developer.android.com/training/articles/perf-jni

The callback only allows to store one pointer. This pointer needs to call Java methods in order to fill the byte buffer.

In order to call the Java methods, we need to have JNIEnv. However, JNIEnv that we have in the method call, won't be valid anymore outside of this call. Therefore, we need to create it ourselves. Attaching it using JavaVM and the current thread is a most common way of doing this. We will also use the method in destructor later.